search-ui/src/pages/Details/Details.js

```diff
@@ -34,37 +38,116 @@ export default function Details() {
 
   }, [id]);
 
+  useEffect(() => {
+    setHighlight(null);
+  }, [q]);
+
+  function GetTagsHTML(tags) {
+
+    if (!!tags) {
+      let tagsHtml = tags.map((tagValue, index) => {
+        if (index < 10) {
+
+          if (tagValue.length > 30) { // check tag name length
+            // create substring of tag name length if too long
+            tagValue = tagValue.substring(0, 30);
```

Comment:
Why did you use substring here and slice earlier?  (Doesn't really matter, minor consistency thing and curiosity on my part.)  I found this: https://masteringjs.io/tutorials/fundamentals/substring#:~:text=The%20slice%28%29%20function%20is%20less%20common%20than%20substring%28%29,parameters%2C%20and%20is%20not%20considered%20a%20legacy%20function.

Reply:
Thanks for that link and sorry for the inconsistency. Based on that page, slice is the best bet so I'll standardize on that